### PR TITLE
Add initial documentation for Blimp

### PR DESCRIPTION
**Pull request by MichelleRos**
This adds some initial documentation for the Blimp vehicle.

### CODE REVIEW

File: blimp/source/docs/getting-started.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. _getting-started:
+
+====================================
+Getting Started
+====================================
+
+This article provides an overview of the main components you will need
+when building and using an Ardupilot-based blimp.
```
Comment (nkruzan) [inline blimp/source/docs/getting-started.rst:8]:
```suggestion
when building and using an ArduPilot-based blimp.
```

File: blimp/source/docs/getting-started.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ArduPilot-capable blimp
+=======================================================
+
+As there are currently no commercially-available ArduPilot-compatible blimps
+available, you will need to :ref:`build your own <building-a-blimp>`.
+
+6+ channel RC transmitter and receiver
+======================================
+
+You'll need a radio control transmitter to manually control your BLimp
```
Comment (nkruzan) [inline blimp/source/docs/getting-started.rst:19]:
```suggestion
You'll need a radio control transmitter to manually control your Blimp
```

File: blimp/source/docs/getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+and to activate its flight modes. You can use any RC
+transmitter/receiver system with at least six channels. Some of the
+options are discussed in the topic :ref:`Compatible RC Transmitter and Receiver Systems <common-rc-systems>`.
+
+.. image:: ../../../images/spektrum-dx8.jpg
+    :target: ../_images/spektrum-dx8.jpg
+
+Autopilot (Autopilot hardware)
```
Comment (nkruzan) [inline blimp/source/docs/getting-started.rst:27]:
```suggestion
Autopilot (hardware)
```

File: blimp/source/docs/getting-started.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+==============================
```
Comment (nkruzan) [inline blimp/source/docs/getting-started.rst:28]:
```suggestion
====================
```

File: blimp/source/docs/getting-started.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+payload added.
+
+While a blimp generally uses only one battery at a time, we recommend having a few batteries in stock; more batteries means more flight time. You'll also need a charger for your batteries.
+
+Ground Control Station (GCS)
+============================
+
+The (free and open source) :ref:`Mission Planner <planner:home>` is recommended if you're going
+to be loading new versions of Blimp onto the autopilot, and for
+first-flight tuning and calibration. It runs on a PC.
```
Comment (nkruzan) [inline blimp/source/docs/getting-started.rst:72]:
```suggestion
first-flight tuning and calibration.
```
It also run on andriod tablets & phones.

**Review by nkruzan — COMMENTED** [review]
Just a few nitpicks, Thanks. Was interesting to read.

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
Somehow ALL the common files got added, not just the ones you added the site directive to....causing these build errors:
there will probably be more once these are fixed since they can nest

/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-ac2_simple_geofence.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-all-vehicle-types.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-antenna-design.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-antenna-tracking.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-apm25-and-26-overview.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-auxiliary-functions.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-bootloader-update.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-buttons.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-camera-control-board.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-camera-trigger-skysight-mono.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-choosing-a-flight-controller.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-choosing-a-ground-station.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-companion-computers.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-disabling-the-built-in-gps-on-apm2-to-use-an-external-gps-instead.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-edge-quick-start.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-erle-brain2-wiring-quick-start.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-external-magnetometer-for-improved-performance.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-fettec-onewire.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-flight-controller-wiring.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-geofencing-landing-page.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-ie24-fuelcell.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-ie650-fuelcell.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-installing-3dr-ublox-gps-compass-module.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-mission-planner-gps-based-antenna-tracking.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-mounting-the-flight-controller.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-mounting-the-gps-module.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-navio-overview.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-navio2-wiring-and-quick-start.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-optional-hardware.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-polygon_fence.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-prearm-safety-checks.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-px4fmu-overview.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-pxfmini-wiring-quick-start.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-qualcomm-snapdragon-flight-kit.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-rcoutput-mapping.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-richenpower-generator.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-simulation.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-ublox-gps.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-use-cases-and-applications.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-using-a-current-sensor.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-vrgimbal.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-web-apps.rst: WARNING: document isn't included in any toctree
/home/henry/Desktop/ardupilot_wiki/blimp/source/docs/common-when-problems-arise.rst: WARNING: document isn't included in any toctree

**Comment by TunaLobster** [comment]
All those warnings are from adding blimp to `DEFAULT_COPY_WIKIS`. There is also a custom tag we created for including the .rst files one at a time. Check out that regex here and from the scattered around tags. https://github.com/ArduPilot/ardupilot_wiki/blob/master/update.py#L406

There are also warnings in CI going the other direction on inclusion of files.
```
2022-03-18T04:26:34.0724785Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-ek3-affinity-lane-switching'
2022-03-18T04:26:34.0725724Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-flight-time-recorder'
2022-03-18T04:26:34.0726593Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-lua-scripts'
2022-03-18T04:26:34.0727471Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-MAVLink2-signing'
2022-03-18T04:26:34.0730497Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-sensor-offset-compensation'
2022-03-18T04:26:34.0733876Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/common-advanced-configuration.rst:10: WARNING: toctree contains reference to nonexisting document 'docs/common-serial-passthrough'
```

There are also syntax warnings:
```
2022-03-18T04:26:34.0720507Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/building-a-blimp.rst:70: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2022-03-18T04:26:34.0721572Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/building-a-blimp.rst:81: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2022-03-18T04:26:34.0722923Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/building-a-blimp.rst:90: WARNING: Explicit markup ends without a blank line; unexpected unindent.
2022-03-18T04:26:34.0723685Z /home/runner/work/ardupilot_wiki/ardupilot_wiki/blimp/source/docs/building-a-blimp.rst:101: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```
It looks like you are wanting captions. I would suggest using the figure directive for that rather than image. https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure

Does there need to be another file created on the server for this file?:
`intersphinx inventory 'https://ardupilot.org/blimp/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: 404 Client Error: Not Found for url: https://ardupilot.org/blimp/objects.inv`

CI seems to indicate that the build completes at least.

**Comment by MichelleRos** [comment]
@Hwurzburg Actually I was a bit unsure what to do with those - we'll probably want many/most of the common docs included with Blimp's documentation eventually (since they mostly concern things that are available to all vehicles) so I did add it to the default copy wiki list, but I just ended up not having the time to deal with adding each of them to a toc tree & figuring out how to structure it... 
So is it better to just remove Blimp from the default copy wiki list and individually add the ones I'm currently using, or how would you want to deal with them?

The blimp/objects.inv bit is something I asked about on Discord a while ago, but ended up not finding a solution to it, so I still have no idea what the file is or how to add it. I've never worked with rst files or sphinx etc, so I've just figured out as much as I can but now I'm basically stuck.

**Comment by MichelleRos** [comment]
Thanks @nkruzan I've added your fixes.

File: blimp/source/docs/getting-started.rst (hunk 1 above)
Comment (MichelleRos) [reply blimp/source/docs/getting-started.rst]: Fixed.

File: blimp/source/docs/getting-started.rst (hunk 2 above)
Comment (MichelleRos) [reply blimp/source/docs/getting-started.rst]: Fixed.

File: blimp/source/docs/getting-started.rst (hunk 3 above)
Comment (MichelleRos) [reply blimp/source/docs/getting-started.rst]: Fixed.

File: blimp/source/docs/getting-started.rst (hunk 4 above)
Comment (MichelleRos) [reply blimp/source/docs/getting-started.rst]: Fixed.

File: blimp/source/docs/getting-started.rst (hunk 5 above)
Comment (MichelleRos) [reply blimp/source/docs/getting-started.rst]: Fixed.

**Comment by Hwurzburg** [comment]
@MichelleRos When I a get some time soon, I can work on getting this site into shape for inclusion...
@peterbarker who creates the objects inventory for a site...or will it magically appear when this gets pushed to the server and it will build them for others to use (my guess)

**Comment by Hwurzburg** [comment]
@TunaLobster @peterbarker so I started cleaning up this for build errors and pushed a version with no errors (except for a duplicated param which I am submitting a PR for, as well as removing some unused ones)....but its very strange...a lot of the ref links are to common docs that are NOT included, yet it gives no warning, build the site locally, and strangely links those missing links to ROVER documents!!! we need to understand this....for example in getting started look at the link to choosing an autopilot...

but it also brings up an interesting way that I could approach this site as minimalist (as requested by Tridge), by reverting my change and just using direct URLs to other site's common pages where needed....the downside is that one leaves the Blimb site and loses the left menu for Blimb....would have to use back arrows to get back to Blimb...so I dont like that approach

anyway, we need to understand the why's of what I am seeing and determine how to get an objects.inv created (may be involved with what se are seeing??)

**Comment by Hwurzburg** [comment]
okay got all the build errors fixed....I think....going to merge and then start working on getting the information completed and formatting cleaned up a bit...fingers crossed

**Comment by MichelleRos** [comment]
Thank you @Hwurzburg 😄

**Pull request merged**